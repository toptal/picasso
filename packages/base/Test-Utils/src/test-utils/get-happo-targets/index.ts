import { getCheckpoints } from './get-checkpoints'
import { getHappoTargets } from './get-happo-targets'

const checkpoints = getCheckpoints()

export { getHappoTargets, getCheckpoints }

export const HAPPO_TARGETS = getHappoTargets(checkpoints)
