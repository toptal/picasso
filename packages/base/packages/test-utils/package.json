{
  "name": "@toptal/picasso-test-utils",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {}
}