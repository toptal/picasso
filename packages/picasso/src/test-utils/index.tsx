import type { ReactElement } from 'react'
import React from 'react'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'

import TestingPicasso from './TestingPicasso'

export type PicassoConfig = {
  titleCase: boolean
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  picassoConfig?: PicassoConfig
): RenderResult => {
  const { rerender, ...rest } = render(
    <TestingPicasso titleCase={picassoConfig?.titleCase}>{ui}</TestingPicasso>,
    options
  )

  const customRerender: typeof rerender = newUi =>
    rerender(
      <TestingPicasso titleCase={picassoConfig?.titleCase}>
        {newUi}
      </TestingPicasso>
    )

  return { rerender: customRerender, ...rest }
}

export * from '@testing-library/react'
export { customRender as render, TestingPicasso }
export { getCheckpoints } from './get-checkpoints'
