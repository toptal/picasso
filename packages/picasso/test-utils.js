// This file is here for the sake of compatability to allow TS versions below 5
// to import from "@toptal/picasso/utils"
// When possible, import directly from "@toptal/picasso-test-utils"
export * from '@toptal/picasso-test-utils'
