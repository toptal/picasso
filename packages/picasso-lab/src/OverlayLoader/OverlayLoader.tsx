import React, { forwardRef } from 'react'
import { Theme, makeStyles } from '@material-ui/core/styles'
import cx from 'classnames'
import { Loader, Container, LoaderProps } from '@toptal/picasso'

import styles from './styles'

type OverlayPosition = 'top' | 'center'

export interface Props extends Omit<LoaderProps, 'inline'> {
  /** Position of the loader on the overlay */
  position?: OverlayPosition
}

const useStyles = makeStyles<Theme, Props>(styles, {
  name: 'PicassoOverlayLoader'
})

export const OverlayLoader = forwardRef<HTMLDivElement, Props>(
  function OverlayLoader(props, ref) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { className, position, ...loaderProps } = props
    const classes = useStyles(props)

    return (
      <Container
        flex
        padded='small'
        ref={ref}
        className={cx(classes.root, className)}
      >
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <Loader {...loaderProps} />
      </Container>
    )
  }
)

OverlayLoader.defaultProps = {
  size: 'medium',
  position: 'center'
}

OverlayLoader.displayName = 'OverlayLoader'

export default OverlayLoader
