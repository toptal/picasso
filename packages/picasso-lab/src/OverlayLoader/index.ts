import { OmitInternalProps } from '@toptal/picasso-shared'

import { Props } from './OverlayLoader'

export { default } from './OverlayLoader'
export type OverlayLoaderProps = OmitInternalProps<Props>
