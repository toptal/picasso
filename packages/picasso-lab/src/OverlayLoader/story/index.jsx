import { OverlayLoader } from '../OverlayLoader'
import PicassoBook from '~/.storybook/components/PicassoBook'

const page = PicassoBook.section('Lab').createPage(
  'OverlayLoader',
  'Use to OverlayLoader user about important information'
)

page.createTabChapter('Props').addComponentDocs({
  component: OverlayLoader,
  name: 'OverlayLoader'
})

page
  .createChapter()
  .addExample('OverlayLoader/story/Default.example.tsx', 'Default')
  .addExample('OverlayLoader/story/Size.example.tsx', 'Size')
  .addExample('OverlayLoader/story/Position.example.tsx', 'Position')
