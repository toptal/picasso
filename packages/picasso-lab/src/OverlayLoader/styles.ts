import { createStyles, Theme } from '@material-ui/core/styles'
import { alpha } from '@toptal/picasso-shared'

import { Props } from './OverlayLoader'

export default ({ palette }: Theme) =>
  createStyles({
    root: {
      position: 'absolute',
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: 'flex',
      justifyContent: 'center',
      alignItems: ({ position }: Props) =>
        position === 'top' ? 'flex-start' : 'center',
      backgroundColor: alpha(palette.common.white, 0.7)
    }
  })
