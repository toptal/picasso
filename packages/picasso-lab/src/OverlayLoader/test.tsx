import React, { ReactNode } from 'react'
import { render } from '@toptal/picasso/test-utils'
import { OmitInternalProps } from '@toptal/picasso-shared'

import OverlayLoader, { Props } from './OverlayLoader'

const renderOverlayLoader = (
  children: ReactNode,
  props: OmitInternalProps<Props>
) => {
  return render(<OverlayLoader size={props.size}>{children}</OverlayLoader>)
}

describe('OverlayLoader', () => {
  test('default render', () => {
    const { container, queryByText } = renderOverlayLoader('Loading data...', {
      size: 'small'
    })

    expect(queryByText('Loading data...')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
