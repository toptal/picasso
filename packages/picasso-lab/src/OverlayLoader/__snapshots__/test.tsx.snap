// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayLoader default render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <div
      class="Container-smallPadding Container-flex PicassoOverlayLoader-root PicassoOverlayLoader-root"
    >
      <div
        class="Loader-root"
      >
        <div
          class="MuiCircularProgress-root CircularProgress-root Loader-spinnerBlue MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 16px; height: 16px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
        <div
          class="Loader-label"
        >
          Loading data...
        </div>
      </div>
    </div>
  </div>
</div>
`;
